rs/crypto/internal/crypto_lib/threshold_sig/bls12_381/src/types/conversions.rs

```diff
@@ -32,29 +33,32 @@ impl PublicKey {
     /// Deserializes a `PublicKey` from a *trusted* source.
     ///
     /// # Security Notice
-    /// This uses the "unchecked" G2 deserialization (no subgroup check),
-    /// so should only be used on `PublicKeyBytes` obtained
-    /// from a known, trusted source.
+    ///
+    /// This function may use the "unchecked" G2 deserialization (no
+    /// subgroup check), so should only be used on `PublicKeyBytes`
+    /// obtained from a known, trusted source.
     pub fn from_trusted_bytes(
         bytes: &PublicKeyBytes,
     ) -> Result<Self, ThresholdSigPublicKeyBytesConversionError> {
-        G2Projective::deserialize_unchecked(&bytes.0)
+        G2Affine::deserialize_cached(&bytes.0)
             .map_err(|_| ThresholdSigPublicKeyBytesConversionError::Malformed {
                 key_bytes: Some(bytes.0.to_vec()),
                 internal_error: "Invalid public key".to_string(),
```

Comment:
Note to self:
* Load transcript and retain only work with serialized transcripts as they only compute the keyID. 
* Also create resharing dealing does not use it.